lib/_http_incoming.js

```diff
@@ -315,6 +315,8 @@ IncomingMessage.prototype._dump = function _dump() {
   if (!this._dumped) {
     this._dumped = true;
     this.resume();
+    // Flush the existing buffered data
+    this._readableState._flush();
```

Comment:
I think the general idea is to *at least* avoid incorporating *new* direct uses of `_*State`. Kicking the can down the road doesn't seem like a good idea IMHO.